### PR TITLE
Added Charge and GPS Home options

### PR DESCRIPTION
**Pull request by S0urceror**
Hi,

I added the Charge state so that I am able to switch charging on/off at home. Sometimes I wish to start it a bit earlier then scheduled charging time. Plus I am recording my electricity use at home at charge start and end.

Also added an option to determine if the Tesla is within 100m of home. This way we can create an automation when I pull up the drive way. And together with the charge option only count kWh when I'm home.

Regards,

Mario

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (hramos) [inline .vscode/launch.json]: This is a Visual Studio Code specific file, and should probably be kept out of the repository.

**Comment by nfarina** [comment]
Thanks for the PR!

I'm curious about the "home" switch - have you successfully created any automations that are triggered by that switch? I don't believe HomeKit will "poll" the switch state regularly, so I'd be surprised if Siri ever "knew" the car had come home, for instance.

**Comment by S0urceror** [comment]
Hi Nick,

Yes you are right.

Already discovered this currently works only on a polling basis. Automations as a result only trigger if you manually open the Home app. This is of course not wanted but let’s call this the first step.

I have earlier implemented a ZWave plugin for Homebridge that works immediately. By holding on to the homebridge objects, everytime one of the ZWave listeners receives an event, I inform homebridge of the change. I will implement this pattern, whenever I have time, on your plugin. Then arriving with my car at home this will trigger immediately something in the home without me having to open the app.

Hopefully soon I will implement this second step.

Let you know.

Regards,

Mario

**Comment by nfarina** [comment]
Hi Mario!

I've reverted the plugin back to a much simpler state - all the various services that were being added were causing Siri to give wacky orders (for example, unlocking the doors caused the trunks to open as well).

I did add a new switch for charging - so you can say "Turn on the Model 3 Charger" for instance and it will begin charging even if it's not scheduled to charge.

If you do implement the "coming home" functionality in a reliable way, I'd love to see it.

Closing this for now!

**Pull request closed without merging**